src/app/(main)/my-voyage/[teamId]/ideation/ideationService.ts

```diff
@@ -53,23 +59,43 @@ export interface IdeationVoteResponse extends IdeationResponse {
   projectIdeaId: number;
 }
 
+const nextJsRestApiAdapter = new NextJsRestApiAdapter(
+  process.env.NEXT_PUBLIC_API_URL!,
+);
+const ideationApiPort = new IdeationApiAdapter(nextJsRestApiAdapter);
+
+interface NextJsAddIdeationAdapter extends AddIdeationUseCase {
+  execute(
+    props: Required<AddIdeationRequestDto>,
+  ): Promise<AddIdeationResponseDto>;
+}
+
 export async function addIdeation({
   teamId,
   title,
   description,
   vision,
-}: AddIdeationProps): Promise<AsyncActionResponse<AddIdeationResponse>> {
+}: AddIdeationUsecaseDto): Promise<
+  AsyncActionResponse<AddIdeationResponseDto>
+> {
   const token = getAccessToken();
-
-  const addIdeationAsync = () =>
-    POST<AddIdeationBody, AddIdeationResponse>(
-      `api/v1/voyages/teams/${teamId}/ideations`,
+  const addIdeationUseCase: NextJsAddIdeationAdapter = new AddIdeationUseCase(
```

Comment:
We can use the "const { Class } = resolveInjection()" (server side) method to import the class from the tsyringe container and use it here or in a component.
Or we can use "const { walletUseCases } = useInjection()" (client side).

Reply:
Here is a stupid example from another project:

// CLIENT SIDE

	const { classUseCase } = useInjection();

	useEffect(() => {
		getData();
	}, []);

	const getData = async () => {
		const response = await classUseCase().getUserData();
		if (response.success) {
                          // do something
		} else {
			// do something
		}
	};


// SERVER SIDE

  const { classUseCase } = resolveInjection();
  const getData = classUseCase.getUserData();

Reply:
How do we resolve the dependency of that usecase? For example in the add ideation usecase, it needs to inject the ideationApi (type ideationApiPort)

Reply:
as you can see I instantiate it like const addIdeationUseCase = new AddIdeationUseCase(ideationApiPort);